assignments/assignment11/task-manager.js

```diff
@@ -0,0 +1,63 @@
+var Task = require('./task')
+var createTaskManager = function () {
+    tasks = [];
+    create = function (category, title, priority, estimate) {
+        var task = new Task(category, title, priority, estimate);
+        tasks.push(task);
+        return task;
+    }
+    get = function (value) {
+        return tasks[value];
+    }
+    getAll = function (active) {
+        if (active === true) {
+            for (var i = 0; i < tasks.length; i++) {
+                if (tasks[i].estimate === 0)
+                { tasks.splice(i, 1); }
+            }
+        }
+        return tasks;
+    }
+    find = function (matcher) {
+        var arr = [];
+        for (var i = 0; i < tasks.length; i++) {
+            var t = true;
+            var c = true;
+            var lowertitle = tasks[i].title.toLowerCase();
+            var lowercategory = tasks[i].category.toLowerCase();
+            for (var j = 0; j < matcher.length; j++) {
```

Comment:
You don't have to write the matching algorithm yourself. Use the builtin methods.